### DISCUSSION TITLE
Consider format for custom image loader

### DISCUSSION THREAD

**Discussion by timfuhrmann** [vercel/next.js#35016] [Ideas]
### Describe the feature you'd like to request

Currently `ImageLoaderProps` includes `src`, `width` and `quality`. I would love to have another property to handle image formats. 

The default image loader out of the box will fallback to default if necessary - it would be cool achieve the same kind of behavior for my custom loader, which currently has the format hardcoded: 

````typescript
export const contentfulImageLoader = ({ src, width, quality }: ImageLoaderProps) => {
    return `${src}?w=${width}&q=${quality || 75}&fm=webp`;
};
````

This works fine but is missing the fallback from `webp`. Another property telling me when to use what format or whether to use an optimized format would be helpful. 

Let me know if there's already an approach how to handle this.

### Describe the solution you'd like

I can imagine a property that returns the best possible/compatible format, maybe from "Acceptable Formats" list, would be ideal.

**Comment by EliseiNicolae** [comment]
Will be awesome if the logic for compatibility (webp) would be handled by Next.js.
